### PR TITLE
Feature/itrex 495 authzlib

### PR DESCRIPTION
**Pull request by acraea-x**

### CODE REVIEW

File: src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java [hunk 1]
```diff
@@ -1,67 +1,390 @@
...
+ * Authentication: org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+ * Principal: org.springframework.security.oauth2.jwt.Jwt 
+ * Details: org.springframework.security.web.authentication.WebAuthenticationDetails
+ * 
+ * Roles need to start with "ROLE_"
+ * 
+ * further details tbd
+ * 
+ */
+
```
Comment (Well5a) [inline src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java:31]: Remove new line

File: src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java [hunk 2]
```diff
@@ -1,67 +1,390 @@
...
+/**
+ * These are the classes currently used: 
+ * 
+ * Authentication: org.springframework.security.oauth2.server.resource.authentication.JwtAuthenticationToken;
+ * Principal: org.springframework.security.oauth2.jwt.Jwt 
+ * Details: org.springframework.security.web.authentication.WebAuthenticationDetails
+ * 
+ * Roles need to start with "ROLE_"
+ * 
+ * further details tbd
```
Comment (Well5a) [inline src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java:28]: open todo?

Comment (Well5a) [inline src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java]: open todo?

File: src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java (hunk 1 above)
Comment (Well5a) [inline src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java:29]: remove new line

File: src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java [hunk 3]
```diff
@@ -1,67 +1,390 @@
...
+     * Checks whether the current user has one of the following {@link CourseRole}s for a given course {@link UUID}:<p>
+     * - <code>RexRole.OWNER</code> <p>
+     * 
+     * @param courseId the {@link UUID} of the course.
+     * @return <code>true</code>/<code>false</code>.
      */
-    public class RexAuthzConstants {
+    public static boolean userIsCourseOwner(UUID courseId) {
+        return userHasCourseRole(courseId, CourseRole.OWNER);
+    }
```
Comment (Well5a) [inline src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java:251]: Wieso keine Funktion für UserIsCourseManager?

File: src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java [hunk 4]
```diff
@@ -1,67 +1,390 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(RexAuthz.class);
+
+    /**
+     * ----- PUBLIC API -----
+     */
+
+    /**
+     * The roles avaiable in ItRex.
+     */
+    public enum RexRole {
```
Comment (Well5a) [inline src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java:42]: move enums to their own class in domain/enumeration

File: src/main/java/de/uni_stuttgart/it_rex/course/web/rest/written/RexAuthzCheckResource.java [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * REST controller for managing {@link User}.
+ */
+@RestController
+@RequestMapping("/api")
+public class RexAuthzCheckResource {
```
Comment (Well5a) [inline src/main/java/de/uni_stuttgart/it_rex/course/web/rest/written/RexAuthzCheckResource.java:38]: please dont merge this file, we don't want to add an endpoint that is not used

**Review by Well5a — CHANGES_REQUESTED** [review]
Please run checkstyle, I get 112 errors in RexAuthz

Also it would be great if we had at least some tests for helper functions in RexAuthz

File: src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java (hunk 2 above)
Comment (acraea-x) [reply src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java]: removed

File: src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java (hunk 1 above)
Comment (acraea-x) [reply src/main/java/de/uni_stuttgart/it_rex/course/service/written/RexAuthz.java:29]: ok

**Review by Well5a — DISMISSED** [review]

**Comment by Well5a** [comment]
fixed things I complained about, please review again.

**Review by Noeliel — APPROVED** [review]

**Review by DasGraMMa — APPROVED** [review]

**Pull request merged**